src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4920,13 +4916,46 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
     return QVariant();
   }
 
+  QgsFeatureRequest request; // TODO only required attributes
+  QString filterString;
+  node = QgsExpressionUtils::getNode( values.at( 2 ), parent );
+  ENSURE_NO_EVAL_ERROR
+  filterString = node->dump();
```

Comment:
For the filter the comment below (see limit) does not apply. We don't want it to be evaluated with the parent context but set it literally for the request, hence the `dump()`.